### PR TITLE
[Agency Dashboard] Implement download data functionality

### LINKED ISSUE #127: [Frontend][Agency Dashboard] Download button
Implement button to download agency published data based off the feed endpoint

Use the "filenames" key of the metric in the published_data endpoint to know which files to download off the `feed` endpoint

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

https://www.loom.com/share/0633bc39badc415ea0df25e20c8e5dea

Uses the new system key endpoint change [here](https://github.com/Recidiviz/recidiviz-data/pull/17347) in order to implement downloading all feed files associated with a given metric on the charts page.

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #127

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/DashboardView.tsx [hunk 1]
```diff
@@ -199,6 +203,22 @@ const DashboardView = () => {
...
+    a.setAttribute("download", `${filename}.txt`);
+    a.click();
+  };
+
+  const downloadMetricData = () => {
+    const metric = agencyDataStore.metricsByKey[metricKey];
+    if (metric) {
+      metric.filenames.forEach((fileName) => {
+        downloadFeedData(metric.system.key, fileName);
+      });
```
Comment (mxosman) [inline agency-dashboard/src/DashboardView.tsx:218]: Future enhancement: I'm sure you're already thinking about this - I wonder if there's an easy way to zip multiple file downloads - though that might need a library. [Probably not necessary right now.]

File: agency-dashboard/src/DashboardView.tsx [hunk 2]
```diff
@@ -199,6 +203,22 @@ const DashboardView = () => {
     DataVizTimeRangesMap[dataVizStore.timeRange]
   );
 
+  const downloadFeedData = async (system: string, filename: string) => {
+    const a = document.createElement("a");
+    a.href = `/feed/${agencyId}?system=${system}&metric=${filename}`;
+    a.setAttribute("download", `${filename}.txt`);
+    a.click();
```
Comment (mxosman) [inline agency-dashboard/src/DashboardView.tsx:210]: nit: could add `a.remove()` after the `a.click()` to clean up

**Review by mxosman — APPROVED** [review]
LGTM, Terry!

File: agency-dashboard/src/DashboardView.tsx (hunk 2 above)
Comment (terryttsai) [reply agency-dashboard/src/DashboardView.tsx:210]: Good call!

**Pull request merged**